suite-native/module-device-settings/src/hooks/useDeviceConnectionGuard.ts

```diff
@@ -0,0 +1,44 @@
+import { useCallback, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+
+import { useNavigation } from '@react-navigation/native';
+
+import { selectIsDeviceConnected } from '@suite-common/wallet-core';
+import {
+    AuthorizeDeviceStackRoutes,
+    DeviceStackRoutes,
+    RootStackParamList,
+    RootStackRoutes,
+    SettingsStackParamList,
+    StackToStackCompositeNavigationProps,
+} from '@suite-native/navigation';
+
+type NavigationProps = StackToStackCompositeNavigationProps<
+    SettingsStackParamList,
+    DeviceStackRoutes.DeviceSettings,
+    RootStackParamList
+>;
+
+export const useDeviceConnectionGuard = () => {
+    const navigation = useNavigation<NavigationProps>();
+
+    const isDeviceConnected = useSelector(selectIsDeviceConnected);
+
+    const redirectToConnectAndUnlockScreen = useCallback(() => {
+        navigation.navigate(RootStackRoutes.AuthorizeDeviceStack, {
+            screen: AuthorizeDeviceStackRoutes.ConnectAndUnlockDevice,
+            params: {
+                onCancelNavigationTarget: {
+                    name: RootStackRoutes.DeviceSettingsStack,
+                    params: { screen: DeviceStackRoutes.DeviceSettings },
+                },
+            },
+        });
+    }, [navigation]);
+
+    useEffect(() => {
+        if (!isDeviceConnected) return redirectToConnectAndUnlockScreen();
+    }, [redirectToConnectAndUnlockScreen, isDeviceConnected]);
+
+    return { isDeviceConnected, redirectToConnectAndUnlockScreen };
```

Comment:
This is still not needed.
```suggestion
    return { isDeviceConnected };
```

Reply:
https://github.com/trezor/trezor-suite/pull/17545/commits/ca533f6fffc6f9da24a65e9657c77935e847e441 done